### ISSUE TITLE
Feature Request: delete_supercache capability

### ISSUE THREAD

**Issue by smarx80** [Automattic/jetpack#38680]
### Impacted plugin

Super Cache

### What

Replace edit_other_posts with a custom capability “delete_supercache” or add a filteror add a nice apply_filter(

### How

In wpsc_admin_bar_render() there is a check on a hardcoded capability ( edit_other_posts )
which by default corresponds to the editor and administrator user levels.
This could be replaced with a custom capability “delete_supercache” or add a nice apply_filter() so this capability could be customized to permit/deny the deletion of the entire cache.

**Comment by monsieur-z** [comment]
@Automattic/heart-of-gold, is this within the scope of your work? I assumed it was based on [this other issue](https://github.com/Automattic/jetpack/issues/38681).

**Comment by donnchawp** [comment]
@monsieur-z yep. That's something we can look at. Using `add_cap()` seems like the best way of adding the capability. ([docs](https://developer.wordpress.org/plugins/users/roles-and-capabilities/))

**Comment by smarx80** [comment]
Hi, is there any update on this topic? Is there something I can do to help/speedup this implementation? 
Thanks, Simone.
